### PR TITLE
Fix random generator creation in Frame ctor

### PR DESCRIPTION
**Pull request by arneboe**
Before, every time a new Frame was created it also created a new random_generator.
Now one random_generator per thread is created and reused.

Also added a very tiny test that ensures that the uuid is generated and not null

### CODE REVIEW

File: src/Frame.hpp [hunk 1]
```diff
@@ -11,6 +12,9 @@
 
 namespace envire { namespace core
 {
+    using boost::uuids::random_generator;
+    using boost::thread_specific_ptr;
+
```
Comment (saarnold) [inline src/Frame.hpp]: I would avoid to use using statements in header files, since every class that includes this header will have the same namespace or type resolutions. A better option would be to use in-class typedefs.

**Comment by saarnold** [comment]
Since the BaseItem has the same issue, what about having a class e.g. envire::random_generator which is a singleton or has a static boost::random_generator and provides the uids?

**Comment by arneboe** [comment]
I didn't know that BaseItem has the same issue. If more than one class has the issue a global random_generator singleton is the better option. I'll implement it :)

I could limit the scope of the using statement to the class. Would that be ok? Otherwise I'll just remove it.

**Comment by saarnold** [comment]
Thanks :+1: 

I would in general avoid to use the using statement in header files. I think when you use it in the class, the class would appear to have all methods of the included classes, which is maybe also not what you want. And using namespaces makes clearer which type is used and where it comes from. IMHO namespaces make code easier to read and understand.

**Comment by arneboe** [comment]
Ok I introduced a RandomGenerator.hpp which provides access to a thread local static random_generator instance.

File: src/RandomGenerator.hpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright 2015, DFKI GmbH Robotics Innovation Center
+ *
+ *  This file is part of the MARS simulation framework.
```
Comment (saarnold) [inline src/RandomGenerator.hpp]: Envire is not a part of MARS

File: src/RandomGenerator.hpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+/**
+ * \file RandomGenerator.hpp
+ * \author Arne Boeckmann
+ */
+
+#ifndef RANDOMGENERATOR_HPP
+#define	RANDOMGENERATOR_HPP
+#include <boost/thread/tss.hpp>
```
Comment (saarnold) [inline src/RandomGenerator.hpp]: Here is an include missing right?

File: src/RandomGenerator.hpp (hunk 2 above)
Comment (arneboe) [inline src/RandomGenerator.hpp]: whoops. I just copied the header from DummyItem. Is there a license header for envire?

**Comment by saarnold** [comment]
And please squash your commits to one. We should avoid blurring the commit history with the development process of something that should be one commit. When doing changes in PR's you can also amend your last commit and force push the changes.

Sry to be make this such a pain..

**Comment by arneboe** [comment]
k, i squashed them and removed the copyright notice. I also added the missing header.

**Comment by saarnold** [comment]
Ok, thanks.
